master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/AbstractMetaManager.java

```diff
@@ -337,19 +297,24 @@ public void restoreMetaFromFile(File file) throws IOException {
             }
           });
 
-      // read workerInfo
-      int workersSize = in.readInt();
-      for (int i = 0; i < workersSize; ++i) {
-        workers.add((WorkerInfo) in.readObject());
-      }
+      workers.addAll(
+          snapshotMetaInfo.getWorkersList().stream()
+              .map(PbSerDeUtils::fromPbWorkerInfo)
+              .collect(Collectors.toSet()));
+
       partitionTotalWritten.reset();
-      partitionTotalWritten.add(in.readLong());
+      partitionTotalWritten.add(snapshotMetaInfo.getPartitionTotalWritten());
       partitionTotalFileCount.reset();
-      partitionTotalFileCount.add(in.readLong());
-      appDiskUsageMetric.restoreFromSnapshot((AppDiskUsageSnapShot[]) in.readObject());
+      partitionTotalFileCount.add(snapshotMetaInfo.getPartitionTotalFileCount());
+      appDiskUsageMetric.restoreFromSnapshot(
+          snapshotMetaInfo.getAppDiskUsageMetricSnapshotsList().stream()
+              .map(PbSerDeUtils::fromPbAppDiskUsageSnapshot)
+              .collect(Collectors.toList()));
```

Comment:
We can use `org.apache.commons.lang3.Streams.toArray` to collect a stream to a array conveniently to avoid converting to `java.lang.List`, it's expensive than array.

Reply:
Done